assets/chat/js/commands.js

```diff
@@ -0,0 +1,241 @@
+  },
+  {
+    name: 'message',
+    description: 'Send a whisper to <nick>.',
+    alias: ['msg', 'whisper', 'w', 'tell', 't', 'notify'],
+  },
+  {
+    name: 'mute',
+    description: 'Stop <nick> from sending messages.',
+    admin: true,
+  },
+  {
+    name: 'open',
+    description: 'Open a conversation with a user.',
+    alias: ['o'],
+  },
+  {
+    name: 'pin',
+    description: 'Pin a message to chat.',
+    alias: ['motd'],
+    admin: true,
+  },
+  {
+    name: 'poll',
+    description: 'Start a poll.',
+    alias: ['vote'],
+  },
+  {
+    name: 'pollstop',
+    description: 'Stop a poll you started.',
+    alias: ['votestop'],
+  },
+  {
+    name: 'postembed',
+    description: 'Post a video embed in chat.',
+    alias: ['pe'],
+  },
+  {
+    name: 'reply',
+    description: 'Reply to the last whisper you received.',
+    alias: ['r'],
+  },
+  {
+    name: 'showpoll',
+    description: 'Show last poll.',
+    alias: ['showvote'],
+  },
+  {
+    name: 'spoll',
+    description: 'Start a sub-weighted poll.',
+    alias: ['svote'],
+  },
+  {
+    name: 'stalk',
+    description: 'Return a list of messages from <nick>.',
+    alias: ['s'],
+  },
+  {
+    name: 'subonly',
+    description: 'Turn the subscribers-only chat mode <on> or <off>.',
+    admin: true,
+  },
+  {
+    name: 'tag',
+    description: "Mark <nick>'s messages.",
+  },
+  {
+    name: 'timestampformat',
+    description: 'Set the time format of the chat.',
+  },
+  {
+    name: 'unban',
+    description: 'Unban <nick>.',
+    admin: true,
+  },
+  {
+    name: 'unhighlight',
+    description: 'Unhighlight <nick>.',
+  },
+  {
+    name: 'unhost',
+    description: 'Remove hosted content from bigscreen.',
+    admin: true,
+  },
+  {
+    name: 'unignore',
+    description: 'Remove <nick> from your ignore list.',
+  },
+  {
+    name: 'unignoreall',
+    description: 'Remove all users from your ignore list.',
+  },
+  {
+    name: 'unmute',
+    description: 'Unmute <nick>.',
+    admin: true,
+  },
+  {
+    name: 'unpin',
+    description: 'Unpin a message from chat.',
+    alias: ['unmotd'],
+    admin: true,
+  },
+  {
+    name: 'untag',
+    description: 'Untag <nick>.',
+  },
+];
+
+export default class ChatCommands {
+  constructor() {
+    this.regularHeader = 'Available commands: \r';
+    this.adminHeader = 'Available admin commands: \r';
+  }
+
+  /**
+   * @param {CommandList} commands
+   * @param {boolean} [admin]
+   * @returns {string[]}
+   */
+  generateAutocomplete(commands, admin = null) {
+    return commands
+      .filter((command) => admin || !command.admin)
+      .map((command) => [
+        `/${command.name}`,
+        ...(command.alias || []).map((alias) => `/${alias}`),
+      ])
+      .flat();
+  }
+
+  /**
+   * @param {CommandList} commands
+   * @returns {{user: string[], admin: string[]}}
+   */
+  generateHelpStrings(commands) {
+    return {
+      user: [
+        this.regularHeader,
+        ...commands
```

Comment:
Is the spread operator necessary here?

Reply:
Yeah, otherwise we get an array in an array, like this: 
![image](https://user-images.githubusercontent.com/41237021/230272920-8ca2d57d-781d-4133-802f-2f95bbd7a06c.png)
And it messes up the output a bit:
![image](https://user-images.githubusercontent.com/41237021/230272975-0740b5c0-e3b5-481c-8530-9813b4035d00.png)

Reply:
Ah, I see what confused me. Let's instead have two methods that return ready-to-print help strings, one for regular user commands and one for admins.